tests/codegen-llvm/pclmulqdq-target-feature-inlining.rs

```diff
@@ -0,0 +1,34 @@
+// Test for pclmulqdq intrinsics inlining across target_feature boundaries
+// Issue: https://github.com/rust-lang/rust/issues/139029
+//
+// When a function with target_feature calls another function without matching
+// target_feature attributes, the intrinsics may not inline properly, leading
+// to function calls instead of direct instructions.
+
+//@ only-x86_64
+//@ compile-flags: -C opt-level=3
+
+#![crate_type = "lib"]
+
+use std::arch::x86_64 as arch;
+
+// CHECK-LABEL: @reduce128_caller
+// CHECK-NOT: call
+// CHECK: call <2 x i64> @llvm.x86.pclmulqdq
+// CHECK: call <2 x i64> @llvm.x86.pclmulqdq
+// CHECK-NOT: call
```

Comment:
The problem is that this can match the `nocallback` attribute. I think we can avoid it by adding an extra `// CHECK: ret` after it.